pilot/tools/generate_config_crd_types.go

```diff
@@ -0,0 +1,70 @@
+// Tool to generate pilot/pkg/config/kube/types.go
+// Example run command:
+// go run pilot/tools/generate_config_crd_types.go --template pilot/tools/types.go.tmpl --output pilot/pkg/config/kube/crd/types.go
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"text/template"
+
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+type ConfigData struct {
+	IstioKind string
+	CrdKind   string
+}
+
+func MakeConfigData(schema model.ProtoSchema) ConfigData {
+	out := ConfigData{
+		IstioKind: crd.KabobCaseToCamelCase(schema.Type),
+		CrdKind:   crd.KabobCaseToCamelCase(schema.Type),
+	}
+	// Tweak to match current naming. This can be changed to meet the new naming convention.
+	if schema.Group == "authenticaiton" {
+		out.IstioKind = crd.KabobCaseToCamelCase(schema.Group + "-" + schema.Type)
```

Comment:
FYI I plan to do a follow up PR to generate type.go in folder apis/group/version to allow duplicate kinds.